### PR TITLE
feat: add user pages in Renku 2.0

### LINKED ISSUE #3199: User pages
Part of https://github.com/SwissDataScienceCenter/renku/issues/3669, https://github.com/SwissDataScienceCenter/renku-ui/issues/3197.

### PR DESCRIPTION
**Pull request by leafty**
Fixes #3199.

Add user pages to Renku 2.0.

Screenshot 2024-06-17 at 10 47 31

Details:
* Add user page at `/v2/users/:username`.
* Add self-redirect page at `/v2/user`.
* Updated `<Pagination>` component.
* Link to project members in the project info sidebar.

Screenshot 2024-06-17 at 10 46 36

/deploy renku=release-0.54.0 renku-data-services=leafty/fix-251-3

### CODE REVIEW

File: client/src/features/usersV2/show/UserShow.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        generatePath(ABSOLUTE_ROUTES.v2.groups.show, { slug: username }),
+        {
+          replace: true,
+        }
+      );
+    }
+  }, [namespace?.namespace_kind, navigate, username]);
+
+  if (!username) {
+    return <LazyNotFound />;
```
[the remark this answers is not in the document]
Comment (leafty) [reply client/src/features/usersV2/show/UserShow.tsx]: Done in 30e004589f2e2e7500dba9a64c6275777b33b820.

File: client/src/features/usersV2/show/UserRedirect.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          username: user.username,
+        }),
+        { replace: true }
+      );
+    }
+  }, [navigate, user?.username]);
+
+  if (!isUserLoggedIn) {
+    return <LazyNotFound />;
+  }
```
[the remark this answers is not in the document]
Comment (leafty) [reply client/src/features/usersV2/show/UserRedirect.tsx]: Done in ffc4e389f57b94afd2abbd1dc9441794f6a62ea3.

File: client/src/features/rootV2/RootV2.tsx [hunk 3]
```diff
@@ -81,6 +83,14 @@ export default function RootV2() {
               </ContainerWrap>
             }
           />
+          <Route
```
[the remark this answers is not in the document]
Comment (leafty) [reply client/src/features/rootV2/RootV2.tsx:86]: I am not sure of the content that would go on that page, so I will keep it as is for now.

[the remark this answers is not in the document]
Comment (ciyer) [reply client/src/features/rootV2/RootV2.tsx:86]: I think that is fine for this PR, but I think this should be considered before the feature is merged to main.

**Review by ciyer — APPROVED** [review]
LGTM!

File: client/src/features/rootV2/RootV2.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (lokijuhy) [reply client/src/features/rootV2/RootV2.tsx:86]: @ciyer I'm not sure I follow - What is the motivation for such a page? Is this fulfilling a specific user need, or is it more of a best practice around URL and website design?

**Pull request merged**